calculate risk to gentrification and flag for different stages

maggiesu0725 opened the pull request.

maggiesu0725 committed:
intial code

maggiesu0725 committed:
calculate risk to gentrification

maggiesu0725 committed:
add flag for different stage off change

maggiesu0725 commented:
@rpitingolo could you take a look at the average income variable? the change seems unbelievably large for some tracts

rpitingolo commented:
@maggiesu0725 which dataset in the vulnerable to gentriciation.sas program should I be looking at?

maggiesu0725 commented:
@rpitingolo merged and changeintime, specifically, when I run the proc means on changeintime, I get hte averagechange of income for washington region during 2000-2016 is 41637, which is high (maybe not?)

maggiesu0725 commented:
@rpitingolo I didn't adjust for inflation.

maggiesu0725 committed:
add macro to adjust for inflation

maggiesu0725 committed:
update macro parameter for dollar convert

maggiesu0725 commented:
@lhendey see %dollar_convert part

lhendey committed:
moved dollar_convert

@maggiesu0725 - the order of the commands in the data step was the issue - you are creating the variable avghhinc_00 after you ran it through the macro.

maggiesu0725 committed:
update white to non hispanic white and use inflation adjusted numbers to calculate cufoff points

maggiesu0725 committed:
categorize tracts by housing market condition

maggiesu0725 committed:
update code for flagging housing market condition

maggiesu0725 committed:
additional code for neighborhood typology

maggiesu0725 commented:
@lhendey @rpitingolo the program for Lisa Bates typology of gentrification analysis. Might need changes, but main part ready for review

lhendey reviewed (changes requested):
@maggiesu0725 - please see comments here and in the word doc i emailed. cc @rpitingolo

maggiesu0725 committed:
calculate housing cost baseline

maggiesu0725 committed:
add hud income category

maggiesu0725 commented on Prog/Vulnerable to gentrification.sas:
```diff
@@ -24,118 +23,223 @@
+
+data changeintime;
+set merged;
+deltarenter = numrenteroccupiedhu_&_years.- sprntoc0;
```
yes

maggiesu0725 commented on Prog/Vulnerable to gentrification.sas:
Do you mean 0.4 instead of 0.44?

lhendey replied:
no - 0.443 instead of .44

maggiesu0725 committed:
alter housing cost baseline program, separate vulnerable to gentrification program by jurisdiction

maggiesu0725 committed:
add jurisdiction flag

rpitingolo committed:
save threshold values as macro

maggiesu0725 committed:
put jurisdiction in macro and output separately

maggiesu0725 committed:
additional code for by jurisdiction threshold

maggiesu0725 commented:
could you take a look at Vulnerable to gentrification_Jurisdiction? I wrote a macro but it's not running, don't know where to debug. @lhendey

maggiesu0725 committed:
merge threshold to dataset

maggiesu0725 committed:
clean up code

lhendey commented:
@maggiesu0725 - is this ready for review?

maggiesu0725 commented:
It generates the results we want, but I haven’t managed to consolidate the program to make it concise and pretty yet. @lhendey

lhendey committed:
clean up program and correct errors

lhendey commented:
@maggiesu0725 please review the changes i made in https://github.com/NeighborhoodInfoDC/RegHsg/pull/9/commits/40eb1df61227d3e37f54aa51379fdac2bf3417b0 

some were for programming efficiency and some were correcting errors.  

i'd like to also have a version that uses median HH income (not median family income) and see what the sensitivity is. if we don't have that in the currect ACS data please go through the process to add it. for 2000 please look it up for the jurisdictions in american fact finder SF3 table HCT012.  

the adjacency process will need to be repeated. i didn't make all the changes in the regionwide program (which we are no longer using, right?)

maggiesu0725 commented:
Thanks, I'll try. @lhendey

maggiesu0725 committed:
update gitignore

ptatian committed:
Merge branch 'master' into analysis-#8

maggiesu0725 committed:
use median HH income and calculate difference

maggiesu0725 commented:
@lhendey alternative calculation by median hh income is updated. I calculated and there seems no difference between the two in terms of the final flag.

lhendey committed:
pulled in 2000 county medians & changed alt calc

lhendey commented:
@maggiesu0725  - i made some changes. please run the adjacency in ARCGIS to complete the typology.

lhendey commented:
@maggiesu0725  - there are two demographic change flags now.

maggiesu0725 committed:
add hh count and merge back adjacency flag

maggiesu0725 committed:
clean up program and export data for mapping

maggiesu0725 commented:
@lhendey I added the HH count and HH under 80% AMI.

lhendey committed:
LH updates to code

lhendey commented:
@maggiesu0725 -  

I decided to use the 2013-17 data so converted all date references to macro vars in code. 

your adjacent flag file is bringing in some tracts outside the COG region please clip before. 

i've added a flag "potentialADJ" - 
if rank2017 ~=. and rank00_17 ~=. then do;
	if  rank2017<=2 & rank00_17 <=2 then potentialADJ=1; else potentialADJ=0;
	end;
else if rank2017=. or rank00_17=. then potentialADJ=.n;  

correct? 

You should use that to then determine adjacency to high 2017 or high2000-2017 tracts. I also need to be able to see that in ARC GIS. I'm not sure from what i can see on L? (to do some QC on the adjacency). '

i added code to calculate the typology for either median family income or median household income. 

i also added some code at the end to output files we could use in tables. 

I think we are really close. If you can re-run the adjacency - using the 2013-17 data and the potentialADJ flag, I can take a look at the ARC GIS file and then we can merge the updated adjacency back in and finalize this. 

Thanks!

maggiesu0725 committed:
minor changes

maggiesu0725 committed:
set up mapping

maggiesu0725 committed:
maps

maggiesu0725 committed:
finish map

lhendey commented:
@maggiesu0725 - i'm getting errors trying to run the R file.

maggiesu0725 committed:
modify border width

lhendey committed:
change to cogregion_sf?

@maggiesu0725  - shouldn't it be COGregion_sf? also still won't run the readcsv

lhendey committed:
update missing for adjacentbase

maggiesu0725 committed:
some update on map

maggiesu0725 committed:
add county boundary shapefile

maggiesu0725 committed:
add new category to not at risk typology

lhendey committed:
Update Vulnerable to gentrification_Jurisdiction.sas

maggiesu0725 committed:
add missing data category to map

maggiesu0725 committed:
update color scheme

maggiesu0725 committed:
update color scheme and categories

maggiesu0725 committed:
adjust label position

maggiesu0725 committed:
update map with with new categorization

maggiesu0725 committed:
update category order

lhendey committed:
fix values for map

maggiesu0725 committed:
clean up code and update pdf/png
